geoutils/georaster/raster.py

```diff
@@ -817,14 +813,142 @@ def is_modified(self) -> bool:
         return self._is_modified
 
     @property
-    def nodata(self) -> int | np.integer | float | np.floating | None:
+    def nodata(self) -> int | float | list[int] | list[float] | None:
         """
-        Get nodata value
+        Get nodata value.
 
-        :returns: nodata value
+        :returns: Nodata value
         """
         return self._nodata
 
+    @nodata.setter
+    def nodata(self, new_nodata: int | float | list[int] | list[float] | None) -> None:
```

Comment:
Is it actually possible to see that docstring?? By calling `help(self.nodata)`? That would be very unusual.
Any way, my thought is that this message is going to be missed, so to avoid problems, I would just raise an error stating that `self.nodata` can only be updated with `self.update_nodata()`. This way, the user will really get the chance to understand what is being done.

Reply:
I think this is already taken care of:

- The `nodata.setter` always performs default behaviour, which is what we need in 95% of cases. 
- For the other 5%, a UserWarning is raised by `set_nodata` who detects something is unusual, and the warning explicity states what will happen and to use `set_nodata` to avoid the default setting.

This is sufficient, no?

Reply:
I made the docstring here for us mostly, and API documentation :smile:

Reply:
Alright, let's try like this.